File: server/src/TUTORA.Domain/Entities/Quiz.cs
using LMS.Domain.Common;

namespace LMS.Domain.Entities;

public class Quiz : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int PassingScore { get; set; } = 70;
    public Guid LessonId { get; set; }

    public Lesson Lesson { get; set; } = null!;
    public ICollection<QuizQuestion> Questions { get; set; } = [];
    public ICollection<QuizAttempt> Attempts { get; set; } = [];
}

File: client/src/types/quiz.ts
export interface Quiz {
  id: string;
  title: string;
  description?: string;
  passingScore: number;
  lessonId: string;
}

export interface QuizDetail extends Quiz {
  questions: QuizQuestion[];
}

export interface QuizQuestion {
  id: string;
  text: string;
  type: "SingleChoice" | "MultipleChoice" | "TrueFalse";
  points: number;
  answers: QuizAnswer[];
}

/** Student-facing answer — backend does not expose isCorrect. */
export interface QuizAnswer {
  id: string;
  text: string;
}

export interface QuizAttempt {
  id: string;
  quizId: string;
  score: number;
  isPassed: boolean;
  attemptedAt: string;
}

export interface SubmitAttemptData {
  answers: { questionId: string; selectedAnswerIds: string[] }[];
}
